src/main/java/com/ericsson/eiffelcommons/subscriptionobject/RestPostSubscriptionObject.java

```diff
@@ -20,9 +20,37 @@ public RestPostSubscriptionObject(String subscriptionName) throws IOException {
      * @param username
      * @param password
      */
+    @Deprecated
     public void setBasicAuth(String username, String password) {
+        setAuthenticationTypeUsernameAndPassword("BASIC_AUTH", username, password);
+    }
+
+    /**
+     * Sets the field authenticationType together with the username and password.
+     * @param authenticationType
+     * @param username
+     * @param password
+     */
+    public void setAuthenticationTypeUsernameAndPassword(String authenticationType, String username, String password) {
+        setAuthenticationType("BASIC_AUTH");
+        setUsernameAndPassword(username, password);
+    }
```

Comment:
First question, shouldnt you use authenticationType variable instead of "BASIC_AUTH" on the row below.Hmm, i'm not sure about.

Second, we should really rethink the structure of the subscription. Username/password should only be existing if we have certain authtypes, for example BASIC_AUTH. If we have some other kind of authtype it might not exist. I don't expect this to be present in this PR, but we should maybe start thinking about that.

Reply:
First question Yes, that I missed. Second part maybe, but sonce this PR also provides possibility to add only authenticationType and not passwords that is a subscription structure discussion.

would not mind having this structure:
authentication: {
   "type": "some authentication type", (mandatory)
   "username": "some_username", (optional)
   "password": "some_password", (optional)
   "token": "some_token" (optional)
}

Reply:
Yes or:

authType: "BASIC_AUTH" (Choose between certain values (BASIC_AUTH, NO_AUTH) etc).
"BASIC_AUTH": ( "BASIC_AUTH" is mandatory if authType == "BASIC_AUTH") {
  "username": "user" (mandatory),
  "password": "pass" (mandatory)
}